grpc/src/main/java/com/linecorp/armeria/server/grpc/AbstractServerCall.java

```diff
@@ -91,33 +88,28 @@
 
 /**
  * Encapsulates the state of a single server call, reading messages from the client, passing to business logic
- * via {@link ServerCall.Listener}, and writing messages passed back to the response.
+ * via {@link Listener}, and writing messages passed back to the response.
  */
-final class ArmeriaServerCall<I, O> extends ServerCall<I, O>
-        implements Subscriber<DeframedMessage>, TransportStatusListener {
+abstract class AbstractServerCall<I, O> extends ServerCall<I, O> {
 
-    private static final Logger logger = LoggerFactory.getLogger(ArmeriaServerCall.class);
-
-    @SuppressWarnings("rawtypes")
-    private static final AtomicIntegerFieldUpdater<ArmeriaServerCall> pendingMessagesUpdater =
-            AtomicIntegerFieldUpdater.newUpdater(ArmeriaServerCall.class, "pendingMessages");
+    private static final Logger logger = LoggerFactory.getLogger(AbstractServerCall.class);
 
     private static final Splitter ACCEPT_ENCODING_SPLITTER = Splitter.on(',').trimResults();
 
     private final MethodDescriptor<I, O> method;
     private final String simpleMethodName;
 
     private final HttpRequest req;
-    private final StreamMessage<DeframedMessage> deframedRequest;
     private final ArmeriaMessageFramer responseFramer;
 
-    private final HttpResponseWriter res;
+    private final HttpResponse res;
     private final CompressorRegistry compressorRegistry;
     private final ServiceRequestContext ctx;
     private final SerializationFormat serializationFormat;
     private final GrpcMessageMarshaller<I, O> marshaller;
     private final boolean unsafeWrapRequestBuffers;
-    private final ResponseHeaders defaultHeaders;
+    @Nullable
+    private final String clientAcceptEncoding;
```

Comment:
We might want to merge @ta7uw's autoCompression PR before merging this in.

Reply:
Thanks for the ping. 😀